src/WebSarcoClient.ts

```diff
@@ -61,18 +69,65 @@ export class WebSarcoClient {
     // TODO: Allow client to choose when to start/stop libp2p node
     await this.startLibp2pNode();
 
+    let bundlrProvider: ethers.providers.Web3Provider;
+
+    if (params.bundlrPublicKey) {
+      // Custom provider for bundlr that uses a Sarcophagus DAO server-provided public key to sign transactions to sponsor
+      // upload costs for embalmers.
+      const pubKey = Buffer.from(params.bundlrPublicKey, 'hex');
+      bundlrProvider = {
+        getPublicKey: async () => {
+          return pubKey;
+        },
+        getSigner: () => {
+          return {
+            publicKey: pubKey,
+            getAddress: () => pubKey,
+            _signTypedData: async (
+              _domain: never,
+              _types: never,
+              message: { address: string; 'Transaction hash': Uint8Array }
+            ) => {
+              let messageData = Buffer.from(message['Transaction hash']).toString('hex');
+              // const res = await fetch('http://localhost:4000/bundlr/signData', {
+              const res = await fetch('https://api.encryptafile.com/bundlr/signData', {
+                method: 'POST',
+                headers: { 'content-type': 'application/json' },
+                body: JSON.stringify({ messageData }),
+              });
+              const { signature } = await res.json();
+              const bSig = Buffer.from(signature, 'hex');
+              // pad & convert so it's in the format the signer expects to have to convert from.
+              const pad = Buffer.concat([Buffer.from([0]), Buffer.from(bSig)]).toString('hex');
+              return pad;
+            },
+          };
+        },
+        _ready: () => {},
+      } as unknown as ethers.providers.Web3Provider;
+    } else {
+      bundlrProvider = new ethers.providers.Web3Provider(this.provider as any);
+    }
+
+    const bundlrConfig = {
+      timeout: 100000,
```

Comment:
Is this timeout related to just getting the signature, or would this also timeout on the bundlr upload itself?